compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt

```diff
@@ -84,3 +83,30 @@ internal fun FqName.isAnvilModule(): Boolean {
 internal fun <T : ClassReference> ClassId.classReferenceOrNull(
   module: ModuleDescriptor
 ): T? = asSingleFqName().toClassReferenceOrNull(module) as T?
+
+/**
+ * Returns the single element matching the given [predicate], or `null` if element was not found.
+ * Unlike [singleOrNull] this method throws an exception if more than one element is found.
+ */
+internal inline fun <T> Iterable<T>.singleOrEmpty(predicate: (T) -> Boolean): T? {
+  var single: T? = null
+  var found = false
+  for (element in this) {
+    if (predicate(element)) {
+      if (found) throw IllegalArgumentException(
+        "Collection contains more than one matching element."
+      )
+      single = element
+      found = true
+    }
+  }
+  return single
+}
+
+private val truePredicate: (Any?) -> Boolean = { true }
```

Comment:
Why not just integrate this into singleOrEmpty? I don't see it used elsewhere.

Reply:
To avoid allocating a lambda every single time.